httpmw/logging/logging.go

```diff
@@ -0,0 +1,71 @@
+package logging
+
+import (
+	"net/http"
+	"time"
+
+	"github.com/gorilla/mux"
+)
+
+type Logger interface {
+	Info(msg string, keysAndValues ...interface{})
+	Error(msg string, keysAndValues ...interface{})
+}
+
+// more info about gorilla/mux mw here:
+// https://github.com/gorilla/mux/tree/db9d1d0073d27a0a2d9a8c1bc52aa0af4374d265?tab=readme-ov-file#middleware
+func NewLogging(logger Logger) mux.MiddlewareFunc {
+	return func(next http.Handler) http.Handler {
+		return &loggingMiddleware{
+			next:   next,
+			now:    time.Now,
```

Comment:
Why register the function? It doesn't seem necessary.

Reply:
By returning func(http.Handler) http.Handler, we are adhering to the mux.Middleware func which is needed to easily register our middleware to a router. The NewLogging function encapsulates the logging fucntionality and allows us to use it with any logger we pass in. 

![image](https://github.com/user-attachments/assets/06000fdd-e2bd-480e-9c60-09e1baf4cf42)

https://github.com/gorilla/mux/tree/db9d1d0073d27a0a2d9a8c1bc52aa0af4374d265?tab=readme-ov-file#middleware

Reply:
I mean why use field now to register the time.Now() function?

Reply:
It allows for better testing since we can mock it, and control the time to test certain behaviors. It also ensures consistency so that the same time function is used everywhere in the middleware. 

The Fleet logging mw utilizes this as well.